software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcessDriverLifecycleEffectorTasks.java

```diff
@@ -169,7 +173,7 @@ protected String startProcessesAtMachine(final Supplier<MachineLocation> machine
     }
 
     @Override
-    protected void postStartCustom() {
+    protected void postStartCustom(AtomicReference<ReleaseableLatch> startLatchRef) {
```

Comment:
There are subclasses of `SoftwareProcessDriverLifecycleEffectorTasks` and `MachineLifecycleEffectorTasks` in brooklyn-library (e.g. https://github.com/apache/brooklyn-library/blob/master/software/cm/ansible/src/main/java/org/apache/brooklyn/entity/cm/ansible/AnsibleLifecycleEffectorTasks.java#L138, which should really have an `@Override` annotation) and elsewhere in the wild so we should at least deprecate the no-arg methods.